mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -435,11 +435,7 @@ def replace(match: T.Match[str]) -> str:
             var = str(match.group(1))
             try:
                 val = _unholder(self.variables[var])
-                if not isinstance(val, (str, int, float, bool)):
```

Comment:
This didn't used to allow list or dict, but after this patch the shared code does. Needs a FeatureNew specifically for this case.

Reply:
Done already